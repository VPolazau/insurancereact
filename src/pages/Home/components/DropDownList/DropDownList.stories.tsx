import { MemoryRouter } from 'react-router-dom';

import { DropDownList } from './DropDownList';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof DropDownList> = {
    title: 'UI/DropDownList',
    tags: ['autodocs'],
    component: DropDownList,
    argTypes: {
        mainLinkPath: {
            type: 'string',
            description: 'Url главной ссылки',
        },
        mainLinkValue: {
            type: 'string',
            description: 'Название главной ссылки',
        },
        className: {
            type: 'string',
            description: 'Имена классов (заменяют стандартные стили)',
        },
        dropNavLinks: {
            type: 'string',
            description: 'dropLinkPath - url адрес, dropLinkValue - название главной ссылки ',
        },
    },
};

export default meta;
type Story = StoryObj<typeof DropDownList>;
export const DropDownLink: Story = {
    args: {
        mainLinkPath: '/',
        mainLinkValue: 'Имущество',
        dropNavLinks: [
            { dropLinkPath: '/', dropLinkValue: 'Квартира' },
            { dropLinkPath: '/', dropLinkValue: 'Дом' },
        ],
    },
    decorators: [
        (Story) => (
            <div style={{ height: '150px' }}>
                <MemoryRouter>
                    <Story />
                </MemoryRouter>
            </div>
        ),
    ],
};
