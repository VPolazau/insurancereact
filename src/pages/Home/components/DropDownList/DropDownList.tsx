import React from 'react';
import cn from 'classnames';

import { NavLink } from 'react-router-dom';

import styles from './styles.module.css';

export interface IDropDownList {
    className?: string;
    mainLinkPath: string;
    mainLinkValue: string;
    dropNavLinks: {
        dropLinkPath: string;
        dropLinkValue: string;
    }[];
}
export const DropDownList: React.FC<IDropDownList> = ({ className, mainLinkPath, mainLinkValue, dropNavLinks }) => {
    return (
        <div className={styles.dropDownListBasic} data-testid="dropDownList">
            <NavLink to={mainLinkPath} className={styles.dropDownMainLink}>
                {mainLinkValue}
            </NavLink>
            <div className={styles.dropDownContent}>
                {dropNavLinks.map(({ dropLinkPath = '/', dropLinkValue }) => (
                    <NavLink key={dropLinkValue} to={dropLinkPath} className={cn(styles.dropDownText, className)}>
                        {dropLinkValue}
                    </NavLink>
                ))}
            </div>
        </div>
    );
};
