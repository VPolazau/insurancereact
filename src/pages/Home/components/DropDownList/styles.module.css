.dropDownListBasic {
    position: relative;
    display: inline-block;
}

.dropDownMainLink {
    padding: 0 0 9px;
    font-family: 'Nunito Sans', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 17px;
    line-height: 130%;
    color: #1F1D1D;
    border-bottom: 1px solid transparent;
}

.dropDownListBasic:hover .dropDownMainLink:after{
    content: '';
    display: block;
    width: 100%;
    height: 1px;
    background-color: #22C062;
    position: absolute;
    margin-top: 10px;
}

.dropDownContent {
    margin-top: 11px;
    display: none;
    position: absolute;
    background-color: #ffffff;
    width: 200px;
    box-shadow: 0 15px 20px rgba(0, 0, 0, 0.05);
    z-index: 1;
    border-radius: 0 0 6px 6px;
}

.dropDownListBasic:hover .dropDownContent {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 20px;
    gap: 8px;
}

.dropDownContent .dropDownText {
    font-family: 'Nunito Sans', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 130%;
    font-feature-settings: 'pnum' on, 'lnum' on, 'liga' off;
    color: #1F1D1D;
}

.dropDownContent .dropDownText:hover {
    color: rgba(30, 177, 89, 1);
}