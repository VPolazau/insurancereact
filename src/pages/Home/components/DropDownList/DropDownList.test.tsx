import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { DropDownList } from './DropDownList';

describe('DropDownList', () => {
    it('check dropdown link-menu', async () => {
        const dropNavLinks = [
            { dropLinkPath: '/', dropLinkValue: 'Квартира' },
            { dropLinkPath: '/', dropLinkValue: 'Дом' },
        ];
        render(
            <MemoryRouter>
                <DropDownList mainLinkPath="/" mainLinkValue="Имущество" dropNavLinks={dropNavLinks} />
            </MemoryRouter>
        );

        fireEvent.mouseOver(screen.getByText('Имущество'));

        await screen.findByTestId('dropDownList');

        expect(screen.getByText('Квартира')).toBeInTheDocument();
        expect(screen.getByText('Дом')).toBeInTheDocument();
        await expect(screen.getByTestId('dropDownList')).toMatchSnapshot();
    });
});
