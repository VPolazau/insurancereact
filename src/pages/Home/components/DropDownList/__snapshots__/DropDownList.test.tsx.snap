// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownList check dropdown link-menu 1`] = `
<div
  class="dropDownListBasic"
  data-testid="dropDownList"
>
  <a
    aria-current="page"
    class="dropDownMainLink active"
    href="/"
  >
    Имущество
  </a>
  <div
    class="dropDownContent"
  >
    <a
      aria-current="page"
      class="dropDownText active"
      href="/"
    >
      Квартира
    </a>
    <a
      aria-current="page"
      class="dropDownText active"
      href="/"
    >
      Дом
    </a>
  </div>
</div>
`;
